src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala

```diff
@@ -259,6 +252,49 @@ trait ChoosesMode {
             Vector()
         }
 
+      val chosenCurrentTourMode: Option[BeamTourMode] = personData.currentTourMode match {
+        case Some(tourMode) => Some(tourMode)
+        case None =>
+          val availablePersonalVehicleModes = availablePersonalStreetVehicles.map(x => x.streetVehicle.mode).distinct
+          val availableFirstAndLastLegModes =
+            availablePersonalVehicleModes.flatMap(x => BeamTourMode.enabledModes.get(x)).flatten
+          val modesToQuery =
+            (availablePersonalVehicleModes ++ BeamMode.nonPersonalVehicleModes ++ availableFirstAndLastLegModes).distinct
+              .intersect(availableModes)
+          val dummyVehicleType = beamScenario.vehicleTypes(dummyRHVehicle.vehicleTypeId)
+          val currentTour = _experiencedBeamPlan.getTourContaining(nextAct)
+          val tourModeCosts = beamServices.skims.od_skimmer.getTourModeCosts(
+            modesToQuery,
+            currentTour,
+            dummyRHVehicle.vehicleTypeId,
+            dummyVehicleType,
+            beamScenario.fuelTypePrices(dummyVehicleType.primaryFuelType)
+          )
+          val modeToTourMode =
+            BeamTourMode.values.map(tourMode => tourMode -> tourMode.allowedBeamModes.intersect(modesToQuery)).toMap
+          val firstAndLastTripModeToTourModeOption = BeamTourMode.values
+            .map(tourMode => tourMode -> tourMode.allowedBeamModesForFirstAndLastLeg.intersect(modesToQuery))
+            .toMap
+          tourModeChoiceCalculator(
+            tourModeCosts,
+            modeChoiceCalculator.modeChoiceLogit,
+            modeToTourMode,
+            Some(firstAndLastTripModeToTourModeOption)
+          )
+      }
+
+      _experiencedBeamPlan.putStrategy(
+        _experiencedBeamPlan.getTourContaining(nextAct),
+        TourModeChoiceStrategy(chosenCurrentTourMode)
+      )
+
+      val availableModesGivenTourMode = getAvailableModesGivenTourMode(
```

Comment:
I think the one other change here is that if the person still chooses a `BIKE_BASED` tour, we cannot let them use a shared bike. Perhaps we need to add another function that is `getAvailableVehiclesGivenTourMode` before this one is called that filters our shared vehicles if we're on a `BIKE_BASED` or `CAR_BASED` tour